tempesta_fw/http_parser.c

```diff
@@ -4180,6 +4185,13 @@ tfw_http_parse_resp(void *resp_data, unsigned char *data, size_t len)
 			/* Status code is fully parsed, move forward. */
 			resp->status = parser->_acc;
 			parser->_acc = 0;
+			/* Some responses are not allowed to have a body. */
+			/* TODO: Add (req == CONNECT && resp == 2xx) */
```

Comment:
This comment is actually part of the processing according to 7230 section 3.3.2. Right now we don't have a link to the paired request at this stage. But I think it doesn't have to be that way. We should be able to 'pop' the paired request as soon as we start processing the response.

Now we only have a link from `req` to `resp` (`req->resp`), as that is dictated by the logic of the current code. When developing the code I actually has a common `pair` member of `TfwHttpMsg` which was pointing to the paired request from the response, and to the paired response from the request. There was no need for that at the time, so a pointer in one direction only had been left. Perhaps this idea can be re-introduced to cover cases like this in the comment, where there's the need to check both the paired request's request type, and the response's status code.

Reply:
Yes, sounds reasonable. But currently there's still no need in paired request since CONNECT method is not supported yet. I will leave that for future updates.